src/EditorFeatures/Test/Diagnostics/TestDiagnosticAnalyzerService.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.Diagnostics.Log;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Diagnostics
+{
+    internal sealed class TestDiagnosticAnalyzerService : DiagnosticAnalyzerService
```

Comment:
@JohnHamby all the IDE diagnostic analyzer tests would now use this test only implementation of DiagnosticAnalyzerService. I would suggest that if you want to locally verify functionality changes for your v2 engine work, you can turn on the IDE v2 Engine in the constructors here and all tests should use the v2 engine.

One category of tests that will fail are those that supply a custom onAnalyzerException delegate. CompilationWithAnalyzers has no public API support to accept such an handler, and currently just logs all analyzer exception diagnostics here (http://source.roslyn.io/#Microsoft.CodeAnalysis/DiagnosticAnalyzer/CompilationWithAnalyzers.cs,53737f78dfb37533,references) and returns them as part of all diagnostics. We would eventually need a public API change to take this delegate as a parameter.